transform/all_collection/spark/utils/utils.py

```diff
@@ -47,3 +46,23 @@ def add_columns(df: DataFrame, cols: Tuple) -> DataFrame:
         df = df.withColumn(c, lit(None))
 
     return df
+
+
+def check_dfs_cols(dfs: Sequence) -> None:
+    """Check if all dataframes have the same column names and schemas.
+    Schema for certain column have to be the same or null type if it is empty.
+    Necessary assumption to merge dataframes and send data to the same solr collection."""
+    df_columns = [df.columns for df in dfs]
+    assert all(
+        (columns == df_columns[0] for columns in df_columns)
+    ), "Dataframes after transformation don't have the same columns name"
+
+    dfs_cols_schema = [[column.dataType for column in df.schema.fields] for df in dfs]
+
+    for column in zip(*dfs_cols_schema):
+        column_schemas = set((str(_col.typeName()) for _col in column))
+        # Columns in all dataframes should have the same type or void + other type
+        if len(column_schemas) == 2:
```

Comment:
I'm not sure if you shouldn't use `is` instead of `==`, but I'm not the python master.

Reply:
Python caches integer objects in the range -5..256 as singleton instances for performance reasons - so it will work in that case, BUT it is not a good practice to relay on that. In that case, we should use the equality operator `==`

Reply:
Hmmm I do not know if I should create some variables for those 2 and 1. The case here is a little bit confusing but in general:
Here we want to check 
- if all columns in each dataframe have the same name as the corresponding column in other dataframes
- if all columns in each dataframe have the same type (`schema length == 1`)(OR if there is a case in which one column in df1 is void type, and the same column in df2 is something different than void - it's also good - `length of schemas == 2`) as the corresponding column in other dataframes

Reply:
I have switched statements a little bit. Do you have some tips to make it cleaner? One line if else statement is too long I guess.
![image](https://user-images.githubusercontent.com/47147210/196399948-7081b8d1-151f-46e7-8b9b-da5a4477222e.png)